.github/workflows/release_parallel.yml

```diff
@@ -164,46 +140,74 @@ jobs:
-                  fetch-depth: 0 # Fetch all history for all branches and tags
+                  fetch-depth: 0
 
-            - name: Merge update branches
+            - uses: ./.github/actions/configure_git
+              with:
+                  gpg_signing_key: ${{ secrets.GPG_SIGNING_KEY }}
+
+            - name: Collect branches
+              id: collect-branches
+              run: |
+                  # Create array of branches
+                  readarray -t BRANCH_ARRAY < <(git branch -r | grep "origin/update-${{ needs.prepare-release.outputs.release-version }}-" | sed 's/origin\///' | xargs -n1)
+
+                  # Output array elements as multiline value
+                  echo "branches<<EOF" >> $GITHUB_OUTPUT
+                  printf '%s\n' "${BRANCH_ARRAY[@]}" >> $GITHUB_OUTPUT
+                  echo "EOF" >> $GITHUB_OUTPUT
+
+            - name: Display collected branches
+              run: |
+                  echo "Collected branches:"
+                  while read branch; do
+                    echo "  - $branch"
+                  done < <(echo "${{ steps.collect-branches.outputs.branches }}")
+                  echo "End of branch list"
+
+            - name: Fetch branches
+              run: |
+                  echo "Fetching all branches..."
+                  BRANCHES_TO_FETCH=""
+                  while read branch; do
+                    BRANCHES_TO_FETCH+=" ${branch}:${branch}"
+                  done < <(echo "${{ steps.collect-branches.outputs.branches }}")
+
+                  echo "BRANCHES_TO_FETCH: $BRANCHES_TO_FETCH"
+                  git fetch origin ${BRANCHES_TO_FETCH}
+
+            - name: Squash changes
               env:
                   PAT: ${{ secrets.LASTMJS_GITHUB_TOKEN }}
               run: |
-                  git config --global user.name 'Jordan Last'
-                  git config --global user.email 'jordan.michael.last@gmail.com'
-                  git config --global commit.gpgsign true
-                  echo -n "$GPG_SIGNING_KEY" | base64 --decode | gpg --import
-                  git config --global user.signingkey C8B77BCBE16CD2B94B43F9C8757397B82D4ED7B0
-
-                  # Determine the current branch name
-                  CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
-                  echo "Current branch: $CURRENT_BRANCH"
-
-                  # Fetch all branches
-                  git fetch --all
-
-                  # List and merge update branches
-                  for branch in $(git branch -r | grep "origin/update-${{ needs.prepare-release.outputs.release-version }}-" | sed 's/origin\///'); do
-                    echo "Merging branch: $branch"
-                    git fetch origin $branch:$branch
-                    git merge --no-ff "$branch" -m "Merge $branch" || {
+                  CURRENT_BRANCH=$(git branch --show-current)
+                  BASE_COMMIT=$(git rev-parse HEAD)
+
+                  while read branch; do
```

Comment:
Look into octopus merge or somehow merging all branches at once (git pull origin b1 b2 b3 etc could work?)

Reply:
I don't think that will be necessary, like I was saying the squash merge is really meant to do just this, you can just keep squashing and then just do one commit at the end without doing anything hacky or weird like a soft reset. It's all very straight forward.